src/sfctl/custom_deployment.py

```diff
@@ -0,0 +1,131 @@
+    for root, _, files in os.walk(directory):
+        for filename in files:
+            if filename.endswith(extension):
+                file_path_list.append(os.path.join(root, filename))
+    return file_path_list
+
+def load_json(file_path):
+    """ Converts the yaml content to json object
+    :param content: Content to be converted to json object
+    """
+    with open(file_path, 'r') as file_pointer:
+        content = file_pointer.read()
+        json_obj = json.loads(json.loads(json.dumps(content)))
+    return json_obj
+
+def deploy_resource(client, resource):
+    """ Deploys the specified resource to cluster
+    :param obj client: Auto generated client
+    :param str resource: Path of the resource file
+    """
+    resource_type = get_resource_type(resource)
+    resource_name = get_resource_name(resource)
+    print("Creating resource: ", resource_name, "of type: ", resource_type.name)
+    if resource_type == ResourceType.application:
+        application_description = load_json(resource)
+        client.mesh_application.create_or_update(resource_name, application_description.get('description')) # pylint: disable=line-too-long
+    elif resource_type == ResourceType.volume:
+        volume_description = load_json(resource)
+        client.mesh_volume.create_or_update(resource_name, volume_description.get('description')) # pylint: disable=line-too-long
+    elif resource_type == ResourceType.network:
+        network_description = load_json(resource)
+        client.mesh_network.create_or_update(resource_name, network_description.get('description').get('name'), network_description.get('description').get('properties')) # pylint: disable=line-too-long
+    elif resource_type == ResourceType.secret:
+        secret_description = load_json(resource)
+        client.mesh_secret.create_or_update(resource_name, secret_description.get('description').get('properties'), secret_description.get('description').get('name')) # pylint: disable=line-too-long
+    elif resource_type == ResourceType.secretValue:
+        secret_value_description = load_json(resource)
+        fully_qualified_resource_name = secret_value_description.get('fullyQualifiedResourceName').split('/') # pylint: disable=line-too-long
+        secret_value_resource_name = fully_qualified_resource_name[1]
+        client.mesh_secret_value.add_value(resource_name, secret_value_resource_name, secret_value_description.get('description').get('name'), secret_value_description.get('description').get('properties').get('value')) # pylint: disable=line-too-long
+    elif resource_type == ResourceType.gateway:
+        gateway_description = load_json(resource)
+        client.mesh_gateway.create_or_update(resource_name, gateway_description.get('description')) # pylint: disable=line-too-long
+    else:
+        raise CLIError('Invalid resource type found %s' %(resource))
+
+def mesh_deploy(client, input_yaml_file_paths, parameters=None):
+    """ This function

```

Comment:
Please add descriptions of the parameters for this function, along with their types

Reply:
done